pkg/controllers/state/suite_test.go

```diff
@@ -1431,3 +1433,132 @@ func ExpectStateNodeNotFoundForMachine(machine *v1alpha5.Machine) *state.Node {
 	ExpectWithOffset(1, ret).To(BeNil())
 	return ret
 }
+
+var _ = Describe("DaemonSet Controller", func() {
+	It("should not update daemonsetCache when daemonset is created", func() {
+		daemonset := test.DaemonSet(
+			test.DaemonSetOptions{PodOptions: test.PodOptions{
+				ResourceRequirements: v1.ResourceRequirements{Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("1Gi")}},
+			}},
+		)
+		ExpectApplied(ctx, env.Client, daemonset)
+		ExpectReconcileSucceeded(ctx, daemonsetController, client.ObjectKeyFromObject(daemonset))
+		daemonsetPod := cluster.GetDaemonSetPod(daemonset)
+		Expect(daemonsetPod).To(BeNil())
+	})
+	It("should update daemonsetCache when daemonset pod is created", func() {
+		daemonset := test.DaemonSet(
+			test.DaemonSetOptions{PodOptions: test.PodOptions{
+				ResourceRequirements: v1.ResourceRequirements{Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("1Gi")}},
+			}},
+		)
+		ExpectApplied(ctx, env.Client, daemonset)
+		daemonsetPod := test.UnschedulablePod(
+			test.PodOptions{
+				ObjectMeta: metav1.ObjectMeta{
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							APIVersion:         "apps/v1",
+							Kind:               "DaemonSet",
+							Name:               daemonset.Name,
+							UID:                daemonset.UID,
+							Controller:         ptr.Bool(true),
+							BlockOwnerDeletion: ptr.Bool(true),
+						},
+					},
+				},
+			})
+		daemonsetPod.Spec = daemonset.Spec.Template.Spec
+		ExpectApplied(ctx, env.Client, daemonsetPod)
+		ExpectReconcileSucceeded(ctx, daemonsetController, client.ObjectKeyFromObject(daemonset))
+
+		storedPod := cluster.GetDaemonSetPod(daemonset)
+		Expect(storedPod).To(Equal(daemonsetPod))
+	})
+	It("should not update daemonsetCache with the same daemonset pod spec", func() {
```

Comment:
What is this meant to test? Why does this matter?

Reply:
That was a test prevent check if the pods were getting changed for the same pods. After thinking more about it. I don't think it's necessary.